R/opts.R

```diff
@@ -32,151 +19,93 @@
 #' @return A list summarising the input delay distributions.
 #' @author Sebastian Funk
 #' @author Sam Abbott
-#' @inheritParams get_generation_time
 #' @seealso convert_to_logmean convert_to_logsd bootstrapped_dist_fit dist_spec
 #' @export
 #' @examples
-#' # default settings with a fixed generation time
+#' # default settings with a fixed generation time of 1
 #' generation_time_opts()
 #'
 #' # A fixed gamma distributed generation time
-#' generation_time_opts(mean = 3, sd = 2)
+#' generation_time_opts(dist_spec(mean = 3, sd = 2, max = 15))
 #'
 #' # An uncertain gamma distributed generation time
-#' generation_time_opts(mean = 3, sd = 2, mean_sd = 1, sd_sd = 0.5)
-#'
-generation_time_opts <- function(..., disease, source, max = 15L,
-                                 fixed = FALSE, prior_weight = NULL) {
-  dot_options <- list(...) ## options for dist_spec
-  ## check consistent options are given
-  type_options <- (length(dot_options) > 0) + ## distributional parameters
-    (!missing(disease) && !missing(source)) ## from included distributions
-  if (type_options > 1) {
-    stop(
-      "Generation time should be given either as distributional options  ",
-      "or as disease/source, but not both."
-    )
-  }
+#' generation_time_opts(dist_spec(mean = 3, sd = 2, mean_sd = 1, sd_sd = 0.5, max = 15))
+#' # A generation time sourced from the ligerature
+#' dist <- get_generation_time(disease = "SARS-CoV-2", source = "ganyani")
+#' generation_time_opts(dist)
+generation_time_opts <- function(dist, prior_weight = NULL) {
 
-  ## check to see if options have been passed in as a list
-  if (length(dot_options) == 1 && is.list(dot_options[[1]])) {
-    dot_options <- dot_options[[1]]
+  if (missing(dist)) {
+    dist <- dist_spec(mean = 1)
```

Comment:
why not just have this as the default?

Reply:
Done in https://github.com/epiforecasts/EpiNow2/pull/363/commits/6977eaa09ec1c9d764d9ed2d0293ed24a9ecba03